src/ResourceManager/EventGrid/Commands.EventGrid/Domain/RemoveAzureEventGridDomain.cs

```diff
@@ -0,0 +1,99 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.EventGrid.Models;
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    [Cmdlet(
+        VerbsCommon.Remove,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventGridDomain",
+        DefaultParameterSetName = DomainNameParameterSet,
+        SupportsShouldProcess = true),
+    OutputType(typeof(bool))]
+
+    public class RemoveAzureEventGridDomain : AzureEventGridCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
@ahamad-MS nit: same comment about removing `ValueFromPipelineByPropertyName` from all cmdlets except `-ResourceId`

Reply:
@cormacpayne  This is also needed otherwise the following scenario will break

Get-AzureRmEventGridDomain -ResourceGroupName aegBugbashResourceGroup2 -Name aegBugBashDomain2 | New-AzureRmEventGridDomainKey -KeyName Key1